packages/angular-material/src/layouts/array-layout.renderer.ts

```diff
@@ -0,0 +1,148 @@
+  setReadonly,
+  UISchemaElement,
+  UISchemaTester,
+  unsetReadonly,
+} from '@jsonforms/core';
+
+@Component({
+  selector: 'app-array-layout-renderer',
+  template: `
+    <div fxLayout="column" fxLayoutGap="16px" [fxHide]="hidden">
+      <mat-toolbar>
+        <h2>{{ this.label }}</h2>
+        <span fxFlex></span>
+        <button
+          mat-icon-button
+          [disabled]="!isEnabled()"
+          (click)="add()"
+          attr.aria-label="{{ 'Add to ' + this.label + 'button' }}"
+        >
+          <mat-icon>add</mat-icon>
+        </button>
+      </mat-toolbar>
+      <p *ngIf="noData">No data</p>
+      <div
+        *ngFor="
+          let item of data;
+          let idx = index;
+          trackBy: trackByFn;
+          last as last
+        "
+      >
+        <mat-card>
+          <mat-card-content>
+            <jsonforms-outlet [renderProps]="getProps(idx)"></jsonforms-outlet>
+          </mat-card-content>
+          <mat-card-actions *ngIf="isEnabled()">
+            <button mat-button (click)="remove(idx)" attr.aria-label="Remove">
+              <mat-icon>remove_circle</mat-icon>
+              Remove
+            </button>
+          </mat-card-actions>
+        </mat-card>
+      </div>
+    </div>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ArrayLayoutRenderer extends JsonFormsArrayControl {
+  noData: boolean;
+  addItem: (path: string, value: any) => () => void;
+  removeItems: (path: string, toDelete: number[]) => () => void;
+  uischemas: {
+    tester: UISchemaTester;
+    uischema: UISchemaElement;
+  }[];
+
+  constructor(private jsonFormAngularService: JsonFormsAngularService) {
+    super(jsonFormAngularService);
+  }
+  remove(index: number): void {
+    this.removeItems(this.propsPath, [index])();
+  }
+  add(): void {
+    this.addItem(this.propsPath, createDefaultValue(this.scopedSchema))();
+  }
+  ngOnInit() {
+    super.ngOnInit();
+    const { addItem, removeItems } = mapDispatchToArrayControlProps(
+      this.jsonFormAngularService.updateCore.bind(this.jsonFormAngularService)
+    );
+    this.addItem = addItem;
+    this.removeItems = removeItems;
+  }
+
+  mapAdditionalProps(props: ArrayControlProps) {
+    this.noData =
+      !props.data || (Array.isArray(props.data) && !props.data.length);
+    this.uischemas = props.uischemas;
+  }
+
+  getProps(index: number): OwnPropsOfRenderer {
+    const uischema = findUISchema(
+      this.uischemas,
+      this.scopedSchema,
+      this.uischema.scope,
+      this.propsPath,
+      undefined,
+      this.uischema
+    );
+    if (this.isEnabled()) {
+      unsetReadonly(uischema);
+    } else {
+      setReadonly(uischema);
+    }
+    return {
+      schema: this.scopedSchema,
+      path: Paths.compose(this.propsPath, `${index}`),
+      uischema
+    };
+  }
+  trackByFn(index: number) {
+    return index;
+  }
+}
+
+export const ArrayLayoutRendererTester: RankedTester = rankWith(
+  4,
+  isObjectArray
```

Comment:
shouldn't we check for `isObjectArrayWithNesting` instead?

Reply:
Isn't that tester too restrictive? If we don't have a nested array or don't provide a detail schema in the options, then the new renderer won't be used.